### PR TITLE
Fixes bug in Series

### LINKED ISSUE #10503: Series return an incorrect result
The latest sympy master:

```
In [1]: f=exp(x**3)*cos(x**6)

In [2]: f.series(x, 0, 14)
Out[2]: 
          6    9    12         
     3   x    x    x      ⎛ 14⎞
1 + x  + ── + ── + ─── + O⎝x  ⎠
         2    6     24         

In [3]: f.series(x, 0, 19)
Out[3]: 
          6    9       12       15        18         
     3   x    x    11⋅x     59⋅x     179⋅x      ⎛ 19⎞
1 + x  + ── + ── - ────── - ────── - ─────── + O⎝x  ⎠
         2    6      24      120       720           
```

### PR DESCRIPTION
**Pull request by mohitacecode**
#### References to other Issues or PRs

#Fixes #10503
#### Brief description of what is fixed or changed
Now, 
`f=exp(x**3)*cos(x**6)`
`f.series(x, 0, 14)`
return correct result.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by mohitacecode** [comment]
closing and reopening to solve the following error

```
curl: (7) Failed to connect to storage.googleapis.com port 443: Connection timed out
Unable to download 3.8 archive. The archive may not exist. Please consider a different version.
```

**Comment by mohitacecode** [comment]
Travis is giving this error:
```
curl: (56) GnuTLS recv error (-54): Error in the pull function.
Unable to download 3.8 archive. The archive may not exist. Please consider a different version.
```

File: sympy/core/function.py [hunk 1]
```diff
@@ -716,17 +716,15 @@ def _eval_nseries(self, x, n, logx):
...
         l = []
+        i = 0
         g = None
-        # try to predict a number of terms needed
-        nterms = n + 2
-        cf = Order(arg.as_leading_term(x), x).getn()
-        if cf != 0:
-            nterms = int(nterms / cf)
-        for i in range(nterms):
+        while len(l) <= n:
```
Comment (jksuom) [inline sympy/core/function.py]: This is inefficient. `n` is not the number of terms but the order (exponent of the order term). For `cos(x**6).series(x, 0, 14)`  this loop would compute 14 non-zero terms though only 2 are needed (3 if also the linear term `0*arg` is counted).

**Comment by jksuom** [comment]
I think that the original code can be used except that the number of terms has to be computed more carefully. It should be the ceiling of `n/cf`, which can be computed as `(n + cf - 1) // cf`.

**Comment by mohitacecode** [comment]
> Travis is giving this error:
> 
> ```
> curl: (56) GnuTLS recv error (-54): Error in the pull function.
> Unable to download 3.8 archive. The archive may not exist. Please consider a different version.
> ```

what about this error?

**Comment by mohitacecode** [comment]
I have done the required changes and It is working fine and I have also added the tests.
@jksuom

File: sympy/core/function.py (hunk 1 above)
Comment (mohitacecode) [reply sympy/core/function.py]: Okay I will done the changes suggested by you.

**Comment by mohitacecode** [comment]
It is failing on :
`n(sqrt(x))._eval_nseries(x, 3, None)`

**Comment by jksuom** [comment]
> sin(sqrt(x))._eval_nseries(x, 3, None)

It is failing because the exponent `cf` of sqrt(x) is 1/2. The formula for the ceiling only works with integer values of `cf`. It seems that the general expression `(n/cf).ceiling()` has to be used.

**Comment by mohitacecode** [comment]
yeah you are right.
Can we use this instead?

```
        if not isinstance(cf,int):
            nterms +=1
```

**Comment by jksuom** [comment]
The formula for `nterms` should create an integer. `.ceiling()` will do that automatically.

**Comment by jksuom** [comment]
Thanks, this is ready.

You may have noticed that essentially the same issue is also present elsewhere. For example,
```
>>> exp(x**3).nseries(x, 0, 14)
1 + x**3 + x**6/2 + x**9/6 + x**12/24 + x**15/120 + x**18/720 + x**21/5040 + x**24/40320 + x**27/362880 + x**30/3628800 + x**33/39916800 + x**36/479001600 + x**39/6227020800 + O(x**42)
```
There are far too many terms. This could be fixed in the same way as what is done in this patch. You may want to open a new PR for a fix.

**Comment by mohitacecode** [comment]
>>> Thanks, this is ready.

Thanks :)

**Comment by mohitacecode** [comment]
> You may have noticed that essentially the same issue is also present elsewhere. For example,
> 
> ```
> >>> exp(x**3).nseries(x, 0, 14)
> 1 + x**3 + x**6/2 + x**9/6 + x**12/24 + x**15/120 + x**18/720 + x**21/5040 + x**24/40320 + x**27/362880 + x**30/3628800 + x**33/39916800 + x**36/479001600 + x**39/6227020800 + O(x**42)
> ```
> 
> There are far too many terms. This could be fixed in the same way as what is done in this patch. You may want to open a new PR for a fix.

I will check it out and will open a new pr very soon.

**Comment by mohitacecode** [comment]
It seems like the for `exp(x**n).nseries(x,0,k)` it is considering it as `exp(x**n).nseries(x,0,k*n)`?

You can se the pattern @jksuom 

```
>>> exp(x**3).nseries(x, 0, 14)
1 + x**3 + x**6/2 + x**9/6 + x**12/24 + x**15/120 + x**18/720 + x**21/5040 + x**24/40320 + x**27/362880 + x**30/3628800 + x**33/39916800 + x**36/479001600 + x**39/6227020800 + O(x**42)

>>> exp(x**2).nseries(x, 0, 14)
1 + x**2 + x**4/2 + x**6/6 + x**8/24 + x**10/120 + x**12/720 + x**14/5040 + x**16/40320 + x**18/362880 + x**20/3628800 + x**22/39916800 + x**24/479001600 + x**26/6227020800 + O(x**28)

>>> exp(x**1).nseries(x, 0, 14)
1 + x + x**2/2 + x**3/6 + x**4/24 + x**5/120 + x**6/720 + x**7/5040 + x**8/40320 + x**9/362880 + x**10/3628800 + x**11/39916800 + x**12/479001600 + x**13/6227020800 + O(x**14)

```

**Comment by mohitacecode** [comment]
Can you guide me please that where to see because I am not able to find the where the issue exactly is in the code.

**Comment by jksuom** [comment]
It is in sympy/functions/elementary/exponential.py: `_eval_nseries()` of `exp`.

**Pull request merged**